cluster-autoscaler/core/scaledown/planner/planner.go

```diff
@@ -274,7 +276,7 @@ func (p *Planner) categorizeNodes(podDestinations map[string]bool, scaleDownCand
 			klog.Warningf("%d out of %d nodes skipped in scale down simulation due to timeout.", len(currentlyUnneededNodeNames)-i, len(currentlyUnneededNodeNames))
 			break
 		}
-		if len(removableList) >= p.unneededNodesLimit() {
+		if len(removableList)-atomicScaleDownNodesCount >= p.unneededNodesLimit() {
 			klog.V(4).Infof("%d out of %d nodes skipped in scale down simulation: there are already %d unneeded nodes so no point in looking for more.", len(currentlyUnneededNodeNames)-i, len(currentlyUnneededNodeNames), len(removableList))
```

Comment:
Let's include `atomicScaleDownNodesCount` in the log.

Reply:
I meant in this line: just report `atomicScaleDownNodesCount` value. Doing it for every considered node is going to result in a lot of log spam.

Reply:
Makes sense. Done.